src/drive/web/modules/services/components/helpers.js

```diff
@@ -2,20 +2,33 @@ import { models } from 'cozy-client'
 
 export const TYPE_DIRECTORY = 'directory'
 
-export const makeNormalizedFile = async (client, folders, file, getIconUrl) => {
+/**
+ * Normalize file for Front usage in <AutoSuggestion> component inside <SearchBar>
+ *
+ * To reduce API call, the fetching of Note URL has been delayed
+ * inside an onSelect function called only if provided to <SearchBar>
+ * see https://github.com/cozy/cozy-drive/pull/2663#discussion_r938671963
+ *
+ * @param client - cozy client instance
+ * @param folders - all the folders returned by API
+ * @param file - file to normalize
+ * @param getIconUrl - method to get icon url
+ * @returns {{path: (*|string), name, icon, id, url: string, onSelect: (function(): Promise<string>)}} file with
+ */
+export const makeNormalizedFile = (client, folders, file, getIconUrl) => {
   const isDir = file.type === TYPE_DIRECTORY
   const dirId = isDir ? file._id : file.dir_id
   const urlToFolder = `${window.location.origin}/#/folder/${dirId}`
 
-  let path, url
+  let path, url, onSelect
   if (isDir) {
     path = file.path
     url = urlToFolder
   } else {
     const parentDir = folders.find(folder => folder._id === file.dir_id)
     path = parentDir && parentDir.path ? parentDir.path : ''
     if (models.file.isNote(file)) {
-      url = await models.note.fetchURL(client, file)
+      onSelect = () => models.note.fetchURL(client, file)
```

Comment:
@acezard : here is the call of the function (unique, in cozy)

Reply:
What do we do with the return of this fetchURL() method? 

Because if we click on the suggestion, it'll call onSelect: https://github.com/cozy/cozy-bar/commit/7b7491dea69d75e8edc15aece2327972f1aae8bf#diff-bd7d5c5cdb7c2b6d78ffe2d7d7045e90beba76f929eff8695969c06a4d1accc7R237 

Then onSelect make the call to fetchURL() but we do we do with the result? There is no window.location or else, how does it works to open the note?

Reply:
ps: 
PRs de fix de cozy-bar en draft:
- https://github.com/cozy/cozy-bar/pull/773
- https://github.com/cozy/cozy-bar/pull/774

Reply:
j'ai ajouté un nouveau commit pour fixer ceci.

Reply:
`metadata` peut être assez volumineux, ça contient les qualifications, les infos EXIF, etc : https://github.com/cozy/cozy-doctypes/blob/master/docs/io.cozy.files_metadata.md
C'est un peu dommage de devoir rapatrier tout le bloc juste pour savoir si c'est une note. Un `file.mime === 'text/vnd.cozy.note+markdown' n'est pas suffisant @Crash-- ?

Et sur un autre sujet, ça m'embête de voir que : 
1. les metadata des notes (content, schema, title, version) sont à la racine de metadata. J'aurais trouvé mieux de grouper tout ça dans un objet `metadata.notes`, ça devient un sacré bazard les metadata.
2. Je ne vois nul part documenté ces attributs : https://github.com/cozy/cozy-doctypes/blob/master/docs/io.cozy.files_metadata.md

Reply:
et un deuxième, car le postMessage serialize la fonction onSelect, je suis donc obligé de réaliser le fetchUrl directement depuis la cozy bar